src/hotspot/share/classfile/classFileParser.hpp

```diff
@@ -561,7 +562,9 @@ class ClassFileParser {
   const Symbol* class_name() const { return _class_name; }
   const InstanceKlass* super_klass() const { return _super_klass; }
 
-  ReferenceType reference_type() const { return _rt; }
+  bool is_instance_ref_klass() const;
+  ReferenceType determine_reference_type() const;
```

Comment:
I'd prefer this was just called `reference_type`.  Like much of the API here, this is about a property of the designated klass.  That it's no longer just a data member accessor is of no particular importance.